mlflow/tracking/fluent.py

```diff
@@ -199,6 +199,98 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         return exc_type is None
 
 
+def start_run_by_name(
+    run_name: str,
+    experiment_id: Optional[str] = None,
+    nested: bool = False,
+    tags: Optional[Dict[str, Any]] = None,
+    description: Optional[str] = None,
+    log_system_metrics: Optional[bool] = None,
+) -> ActiveRun:
+    """
+    Start a new MLflow run, setting it as the active run under which metrics and parameters
+    will be logged. The return value can be used as a context manager within a ``with`` block;
+    otherwise, you must call ``end_run()`` to terminate the current run.
+
+    ``start_run_by_name`` attempts to resume a run with the specified run name and
+    other parameters are ignored.
+
+    Throws an exception if there are multiple runs with the specified run_name
+
+    Args:
+        run_name: Get the run with the specified run_name and log parameters
+            and metrics under that run. The run's end time is unset and its status
+            is set to running, but the run's other attributes (``source_version``,
+            ``source_type``, etc.) are not changed.
+        experiment_id: ID of the experiment under which to create the current run (applicable
+            only when ``run_id`` is not specified). If ``experiment_id`` argument
+            is unspecified, will look for valid experiment in the following order:
+            activated using ``set_experiment``, ``MLFLOW_EXPERIMENT_NAME``
+            environment variable, ``MLFLOW_EXPERIMENT_ID`` environment variable,
+            or the default experiment as defined by the tracking server.
+        nested: Controls whether run is nested in parent run. ``True`` creates a nested run.
+        tags: An optional dictionary of string keys and values to set as tags on the run.
+            If a run is being resumed, these tags are set on the resumed run. If a new run is
+            being created, these tags are set on the new run.
+        description: An optional string that populates the description box of the run.
+            If a run is being resumed, the description is set on the resumed run.
+            If a new run is being created, the description is set on the new run.
+        log_system_metrics: bool, defaults to None. If True, system metrics will be logged
+            to MLflow, e.g., cpu/gpu utilization. If None, we will check environment variable
+            `MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING` to determine whether to log system metrics.
+            System metrics logging is an experimental feature in MLflow 2.8 and subject to change.
```

Comment:
This section needs to be revised as this is not exactly the same as `start_run` function. For example, under experiment_id there is a reference to `run_id` which is not used as a parameter here at all. I would add some description when specific parameters are used or not used, depending on the case (if run_name was found or not found)